### PR TITLE
JBPM-10022 Forms are rendered without Accordion elements when startin…

### PR DESCRIPTION
**Pull request by bxf12315**
…g a process from Business Central

**Thank you for submitting this pull request**

**JIRA**: _(please edit the JIRA link if it exists)_ 

[[link](https://issues.redhat.com/browse/JBPM-10022)](JBPM-10022)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* https://github.com/kiegroup/droolsjbpm-integration/pull/2722
* link 2
* link 3 etc.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by bxf12315** [comment]
Jenkins run fdb

**Review by pefernan — CHANGES_REQUESTED** [review]
Hey @bxf12315 I think I found an issue in your change, effectively it shows the correlation key accordion as in the regular forms displayer, but if I fill the correlation key in the textbox and submit the form. In the process instancess table, it doesn't show the value filled in the form but shows the process instance ID, take a look at the attached gif:
correlationkey

Also I have a concern in the ui, looks like the form is wrapped in a box with the form name inside the Accordion, would it be possible to remove that box and place the form directly inside the accordion?

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins run fdb

> Hey @bxf12315 I think I found an issue in your change, effectively it shows the correlation key accordion as in the regular forms displayer, but if I fill the correlation key in the textbox and submit the form. In the process instancess table, it doesn't show the value filled in the form but shows the process instance ID, take a look at the attached gif: correlationkey
> 
> Also I have a concern in the ui, looks like the form is wrapped in a box with the form name inside the Accordion, would it be possible to remove that box and place the form directly inside the accordion?

Updated

**Comment by bxf12315** [comment]
Jenkins run fdb

File: jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/KieServerFormsStartProcessDisplayer.java [hunk 1]
```diff
@@ -71,10 +72,11 @@ public void init(FormDisplayerConfig<ProcessDefinitionKey, KieServerFormRenderin
         footerButtons.clear();
 
         container.add(formContainer);
-        
-        initDisplayer();
+        correlationKey = new TextBox();
+        correlationKey.setId("kie-server-from-correlationKey");
```
[the remark this answers is not in the document]
Comment (bxf12315) [reply jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/KieServerFormsStartProcessDisplayer.java:76]: I'm not sure it, kieserverforms just has ceate an iframe and get the start form from kie-server, so the iframe's code is generated by kie-server.  if we want to call parent function, we also need windows.parent the key word.

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
@pefernan updated please check it.

**Comment by barboras7** [comment]
jenkins run fdb

**Comment by pefernan** [comment]
jenkins run fdb

**Comment by pefernan** [comment]
jenkins run fdb

**Comment by bxf12315** [comment]
jenkins run fdb

**Comment by barboras7** [comment]
Thanks @bxf12315, the issue seems to be fixed. I'll run the business central on all certifications and will approve the PR tomorrow if I don't come across any issue. Thanks.

**Review by pefernan — APPROVED** [review]
+1 thanks @bxf12315

**Review by barboras7 — APPROVED** [review]

File: jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormRendererProxyServlet.java [hunk 2]
```diff
@@ -107,8 +112,28 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
             String processId = extractValue(processIdPattern, pathInfo);
             ProcessServicesClient processServicesClient = getClient(serverTemplateId,                                                      
                                                                     ProcessServicesClient.class);
-            
-            Long responseBody = processServicesClient.startProcess(containerId, processId, new StringContentMap(body));
+            Long responseBody = 0L;
```
Comment (pefernan) [inline jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormRendererProxyServlet.java:115]: Would you mind add a test the Servlet please?

**Review by pefernan — CHANGES_REQUESTED** [review]

File: jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormRendererProxyServlet.java (hunk 2 above)
Comment (bxf12315) [reply jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormRendererProxyServlet.java:115]: Done

**Review by pefernan — APPROVED** [review]

**Pull request merged**